core/src/main/java/com/bumble/appyx/core/plugin/Plugins.kt

```diff
@@ -21,16 +22,35 @@ interface NodeLifecycleAware : Plugin {
 }
 
 interface UpNavigationHandler : Plugin {
-    fun handleUpNavigation() = false
+    fun handleUpNavigation(): Boolean = false
 }
 
 fun interface Destroyable : Plugin {
     fun destroy()
 }
 
-// TODO: It is not a plugin! Handled only in router!
+/**
+ * Implementing class can handle back presses via [OnBackPressedCallback].
+ *
+ * Implement either [onBackPressedCallback] or [onBackPressedCallbacks], not both.
+ * In case if both implemented, [onBackPressedCallback] will be ignored.
+ * There is runtime check in [Node] to verify correctness.
+ */
 interface BackPressHandler : Plugin {
-    fun onBackPressed()
+
+    val onBackPressedCallback: OnBackPressedCallback? get() = null
```

Comment:
What do you think about just removing this property? 
If we do so we can just remove the checks in the node and there would be no risk isn't it?
Probably the common user will not usually work around this interface as we are already hiding this complexity from  the BaseRoutingSource, so i will say that we can avoid  so i think that we can be less kind with this api and force extensions to pass a list if the want to override.

What do you think?

Reply:
It might be used not only for routing sources but also for interactors.